apps/microtvm/ethosu/src/tvm_ethosu_runtime.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "tvm_ethosu_runtime.h"
+
+#include <ethosu_driver.h>
+
+int32_t TVMEthosULaunch(struct ethosu_driver* driver, void* cms_data, size_t cms_data_size,
+                        uint64_t* base_addrs, size_t* base_addrs_size, int num_tensors) {
+  int32_t result =
+      ethosu_invoke(driver, cms_data, cms_data_size, base_addrs, base_addrs_size, num_tensors);
+
+  // Map errors in invoke to TVM errors
+  if (result != 0) {
+    return -1;
```

Comment:
Are you referring to the CRT `kError` constants? I was keeping it runtime-agnostic and mapping it similar to the normal error handling in operators:
https://github.com/apache/tvm/blob/7c3283cb8a5da199d698f435754b24181c0322d2/src/target/source/codegen_c_host.cc#L253-L276

I don't think that's the shared CRT/C++ approach?

Reply:
Ah, further to my above explanation, the invoke may return a value which is incompatible with TVM - so that should be catered for here explicitly in the invoke <-> TVM glue. For example, if we had add a success return value of `404` from invoke, then this can equally convert `404` to `0` which is the TVM internal contract.

Reply:
right, i think adopting the != 0 convention is fine here. just was curious whether you had a TODO to try and map into the error space or if the comment was descriptive and you wanted to keep as-is. sounds like the latter, which is fine

one thing i had semi-speculatively adopted was: if the top bit is 1, the error is a system-specific thing. curious if that is helpful here?

Reply:
Ah, the intention was to document that I was mapping the error explicitly as it's a bit odd otherwise, hopefully we'll have a better example where the values are more interesting at some point.

On the speculative bit, I think operator errors work differently from runtime errors in this case? Mostly because the executor only really needs to know if the operator failed or not, so I think `0` and `-1` are sufficient for that use case rather than adding the rest of the runtime error codes?

Reply:
we don't need to fix this in this PR, but i'm a big believer in not squashing error codes if at all possible. almost every mystery i've ever experienced in an embedded system has been the kind of problem where more data almost always is the most effective way to understand/solve the problem. we can squash for now but it'd be good to think about ways to expose driver error codes. that could be us allocating a new error code in error_codes.h and then stashing additional information somewhere else (perhaps that calls for yet another pointer to passed in? or perhaps the implementation should just be left to the driver and the driver struct should be expected to carry it).